ops/main.py

```diff
@@ -42,6 +43,114 @@
 logger = logging.getLogger()
 
 
+@dataclasses.dataclass(frozen=True)
+class _JujuContext:
+    # source: https://juju.is/docs/juju/charm-environment-variables
+    # the HookVars function: https://github.com/juju/juju/blob/3.6/worker/uniter/runner/context/context.go#L1398
+    # only allow accesing these vars because these are the currently used vars in main.py
+
+    # env var JUJU_ACTION_NAME
+    action_name: Optional[str] = None
+
+    # env var JUJU_ACTION_UUID
+    action_uuid: Optional[str] = None
+
+    # env var JUJU_CHARM_DIR
+    charm_dir: Optional[str] = None
```

Comment:
Ah yes, this would be much better, since `context.charm_dir` is only used in that function. I can make this change.

After some investigation, I figured that if we remove `_get_charm_dir`, it will break scenario [here](https://github.com/canonical/ops-scenario/blob/main/scenario/ops_main_mock.py#L18). But I think it's still reasonable to remove `_get_charm_dir`, right?

Reply:
> After some investigation, I figured that if we remove `_get_charm_dir`, it will break scenario [here](https://github.com/canonical/ops-scenario/blob/main/scenario/ops_main_mock.py#L18). But I think it's still reasonable to remove `_get_charm_dir`, right?

I think it's fine. We can open PRs to adjust Scenario before the next ops release.